quinn-proto/src/endpoint.rs

```diff
@@ -690,35 +690,27 @@ impl Endpoint {
 
     /// Handle a timer expiring
     pub fn timeout(&mut self, now: u64, conn: ConnectionHandle, timer: Timer) {
+        let mut mux = EndpointMux::new(&mut self.ctx, conn, self.connections[conn.0].side());
+        let closed = self.connections[conn.0].timeout(&mut mux, now, timer);
         match timer {
             Timer::Close => {
                 self.ctx.io.push_back(Io::TimerStop {
                     connection: conn,
                     timer: Timer::Idle,
                 });
                 self.ctx.events.push_back((conn, Event::ConnectionDrained));
```

Comment:
These updates could be moved into `Connection` to be mediated (more concisely) by the mux.

Reply:
On second thought, probably best not to lose track of the `ConnectionDrained` event when we potentially forget the connection below, since it's a useful to the upper layer that this connection's done with.

Actually, come to think of it, `ConnectionDrained` isn't necessary at all, since close timer expiry is always unambiguous. I'll scrap it.